plugin/src/main/kotlin/tanvd/grazi/spellcheck/GraziSpellchecker.kt

```diff
@@ -12,35 +12,39 @@ import com.intellij.spellchecker.tokenizer.TokenConsumer
 import com.intellij.util.Consumer
 import com.intellij.vcs.commit.message.CommitMessageInspectionProfile
 import com.intellij.vcs.commit.message.CommitMessageSpellCheckingInspection
-import org.languagetool.*
+import org.apache.commons.text.similarity.LevenshteinDistance
+import org.languagetool.JLanguageTool
+import org.languagetool.ResultCache
+import org.languagetool.UserConfig
+import org.languagetool.rules.Rule
 import tanvd.grazi.GraziConfig
 import tanvd.grazi.grammar.Typo
 import tanvd.grazi.ide.msg.GraziStateLifecycle
 import tanvd.grazi.language.Lang
-import tanvd.grazi.utils.*
+import tanvd.grazi.utils.Text
+import tanvd.grazi.utils.spellcheckOnly
+import tanvd.grazi.utils.toPointer
+import tanvd.grazi.utils.withOffset
 import tanvd.kex.buildSet
 import tanvd.kex.tryRun
 import java.util.concurrent.TimeUnit
-import kotlin.properties.Delegates
 
 
 object GraziSpellchecker : GraziStateLifecycle {
-    private const val cacheMaxSize = 25_000L
-    private const val cacheExpireAfterMinutes = 5
     private val checkerLang = Lang.AMERICAN_ENGLISH
+    private var checkers: Set<Pair<JLanguageTool, Rule>> = emptySet()
 
     private val ignorePatters: List<(String) -> Boolean> = listOf(Text::isHiddenFile, Text::isURL, Text::isHtmlUnicodeSymbol, Text::isFilePath)
 
-    private fun createChecker(state: GraziConfig.State): JLanguageTool {
-        val cache = ResultCache(cacheMaxSize, cacheExpireAfterMinutes, TimeUnit.MINUTES)
-        // AMERICAN_ENGLISH must always be present
-        return JLanguageTool(checkerLang.jLanguage!!, state.nativeLanguage.jLanguage,
-                cache, UserConfig(state.userWords.toList())).apply {
-            disableRules(allRules.filter { !it.isDictionaryBasedSpellingRule }.map { it.id })
-        }
-    }
+    private val levenshtein = LevenshteinDistance()
```

Comment:
Levenstein now is everywhere. Let's move it to `Text` object :)

Reply:
Done